src/games/ecards.py

```diff
@@ -181,97 +187,68 @@ async def execute_round(self, leader: int, prompt: str, hands: dict):
 
         elif len(shuffled_responses) == 1:
             # Only one person played a card - award them the victory
-            winning_card = shuffled_responses[0]
-            winning_pid = dict_reverse_lookup(cards_played, winning_card)
-            if winning_pid:
-                # send round result
-                await self.channel.send(
-                    embed=self.embed(
-                        f"This round's prompt: \n**{prompt}**\nOnly {self.players[winning_pid]} played a card:\n{winning_card}\nThey win the round by default."
-                    )
+            winning_card, winning_pid = shuffled_responses[0]
+            # send round result
+            await self.channel.send(
+                embed=self.embed(
+                    f"This round's prompt: \n**{prompt}**\nOnly {self.players[winning_pid]} played a card:\n{winning_card}\nThey win the round by default."
                 )
+            )
 
-                # update scoreboard
-                return self._add_score(winning_pid, 1)
-            else:
-                self.logging.warning(
-                    f"{winning_card} not in {cards_played}:: {shuffled_responses}."
-                )
-                return
+            # update scoreboard
+            return self._add_score(winning_pid, 1)
         else:
             # Enough responses for a proper vote
-            end_str = "\n".join((f"- {i}" for i in shuffled_responses))
-
-            em_text = f"This round's prompt: \n**{prompt}**\nThis round's answers:\n{end_str}\nAwaiting choice of a winner from **{self.players[leader]}**."
-            message = await self.channel.send(
-                embed=self.embed(
-                    em_text
+            end_str = "\n".join(
+                (
+                    f"{EMOJI_FORWARD[index + 1]}: {response}"
+                    for index, (response, _) in enumerate(shuffled_responses)
                 )
             )
 
-            # little pause
-            await asyncio.sleep(self.wait_duration)
+            em_text = f"This round's prompt: \n**{prompt}**\nThis round's answers:\n{end_str}\nAwaiting choice of a winner from **{self.players[leader]}**."
+            winner_root_embed = self.embed(em_text)
+            winner_view = CardsSelectWinningPromptView(
+                winner_root_embed,
+                leader,
+                {leader: shuffled_responses},
+                delete_after=True,
+                timeout=31,
+            )
+            await winner_view.send_and_wait(self.channel)
 
-            choice_response = None
+            winner_replies: Dict[int, Tuple[str, int]] = winner_view.responses
```

Comment:
Another note is to define `Tuple[str, int]` as a `CardPid` type alias so it's more obvious what we're doing here. Or even use a CardPid dataclass.